libs/feature-recipe-details/src/lib/feature-recipe-details/feature-recipe-details.component.ts

```diff
@@ -57,4 +71,38 @@ export class FeatureRecipeDetailsComponent {
       },
     );
   }
+
+  openDeleteConfirmationDialog(): void {
```

Comment:
I think both of the below methods can go to their own _helper_ service that could be created next to ui-confirmation-dialog component.
 
```ts
class DeleteDialogService {
  private dialog = inject(MatDialog);

  openConfirmationDialog(data: ConfirmationDialogComponentData) {
    return this.dialog.open<
      UiConfirmationDialogComponent,
      ConfirmationDialogComponentData,
      boolean
    >(UiConfirmationDialogComponent, {
      data
    }).afterClosed();
  }
}

class FooComponent {
  private deleteDialogService = inject(DeleteDialogService);

  openDeleteConfirmationDialog() {
    const { id, recipeName }  = this.recipe();
    this.deleteDialogService.openConfirmationDialog({
        text: `This will delete ${recipeName} and all its ingredients. This action cannot be undone.`,
        header: `Delete ${recipeName}?`,
        isDeleteConfirmation: true,
    }).pipe(
      tap((val) => val && this.store.dispatch(RecipesUiActions.deleteRecipe({ recipeId }));),
      take(1)
    ).subscribe();
  }
}
```

Reply:
Good idea with the service. 

I have some concerns about `val && this.store.dispatch(RecipesUiActions.deleteRecipe({ recipeId }));` though. I don't think the `&&` operator should be used for short-circuiting. I find the `if` statement more readable and understandable. 

```ts
if(val) {
   this.store.dispatch(RecipesUiActions.deleteRecipe({ recipeId }));
}
```

Using short-circuiting makes code difficult to read. An if statement clearly communicates that the code should only be run if a condition is fulfilled. But using a logical operator for the same purpose is just confusing.

What do you think @DDonochVA?

Reply:
I've created a generic `UiConfirmationDialogService` which now handles the opening of the dialog. It also returns an `Observable`.